### PR TITLE
added FoldersFragment

### PR DESCRIPTION
**Pull request by frostyshadows**
Added a Fragment that displays folders based on [Shavonne's designs](https://xd.adobe.com/spec/fee73f20-fd68-4796-8f1a-9ab275e84837/#screen/f7b096b2-e19b-487a-b101-ef0bbc7961a6/LaunchNote%20Photo%20Gallery%20(Class)%20%E2%80%93%201).

### CODE REVIEW

File: app/src/main/java/com/example/ubclaunchpad/launchnote/photoBrowser/FoldersAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    constructor(context: Context, folders: List<Folder>) : this() {
+        this.folders = folders
+        this.context = context
+    }
+
+    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+        val folder = folders[position]
+
+        // load first 3 photos in the folder
+        (0..2).filter { folder.picNoteIds.size > it }
```
Comment (vpnda) [inline app/src/main/java/com/example/ubclaunchpad/launchnote/photoBrowser/FoldersAdapter.kt]: (0..min(2,folder.picNoteIds.size))

File: app/src/main/java/com/example/ubclaunchpad/launchnote/edit/PhotoInfoActivity.kt [hunk 2]
```diff
@@ -95,9 +95,54 @@ class PhotoInfoActivity : AppCompatActivity() {
...
         picNoteToEdit.description = description_input.text.toString()
         // todo vpineda we need to create a dropdown for all of the classes or projects!
-        if(folder_input.text.isNotEmpty())
+        if(folder_input.text.isNotEmpty()) {
             picNoteToEdit.folderId = Integer.parseInt(folder_input.text.toString())
 
+            var folderToInsert = Folder()
+
+            // create new Folder
+            LaunchNoteDatabase.getDatabase(this)?.let {
```
Comment (vpnda) [inline app/src/main/java/com/example/ubclaunchpad/launchnote/edit/PhotoInfoActivity.kt]: Can we move this function to a blocking function or to a function that receives a callback. We might need this for other things such as the info panel (probably)

File: app/src/main/java/com/example/ubclaunchpad/launchnote/photoBrowser/FoldersAdapter.kt (hunk 1 above)
Comment (frostyshadows) [reply app/src/main/java/com/example/ubclaunchpad/launchnote/photoBrowser/FoldersAdapter.kt]: good catch, updated

File: app/src/main/java/com/example/ubclaunchpad/launchnote/edit/PhotoInfoActivity.kt [hunk 3]
```diff
@@ -95,13 +95,26 @@ class PhotoInfoActivity : AppCompatActivity() {
         picNoteToEdit.title = title_input.text.toString()
         picNoteToEdit.description = description_input.text.toString()
         // todo vpineda we need to create a dropdown for all of the classes or projects!
-        if(folder_input.text.isNotEmpty())
+        if (folder_input.text.isNotEmpty()) {
             picNoteToEdit.folderId = Integer.parseInt(folder_input.text.toString())
```
Comment (rupertotorres1) [inline app/src/main/java/com/example/ubclaunchpad/launchnote/edit/PhotoInfoActivity.kt:99]: Are we going to only use integer names for folders? Or are we going to change this to strings?

**Comment by rupertotorres1** [comment]
Some things I noticed:
- When you add a picture to a new folder it looks weird. I don't know if it's supposed to be like this or we are going to change it, but the whole row turns grey and the folder id also spans the whole row.
- I added two pictures to the same folder and deleted the one that appeared first. The remaining picture stayed in the center column, but I think it should move as far left as possible.
- In the folder view, you can't enter the folder or press any picture in the folder, or delete folders or the pictures in them (you need to go to "All" to delete).
- After deleting all pictures in a folder, the folder view stays the same (pictures still appear) until I press take photo or add from library. After pressing take photo or add from library, the empty folder remains, but maybe it should be removed from the view if it has nothing in it.
- Some of these bugs are probably also causing other bugs, because just by playing around I've had photos from one folder randomly appear in another folder (I've seen this happen when I add two pictures, remove the first one  and then add another picture to a new folder)

File: app/src/main/java/com/example/ubclaunchpad/launchnote/edit/PhotoInfoActivity.kt (hunk 3 above)
Comment (rupertotorres1) [reply app/src/main/java/com/example/ubclaunchpad/launchnote/edit/PhotoInfoActivity.kt:99]: nvm, just saw the TODO

**Review by rupertotorres1 — CHANGES_REQUESTED** [review]
Added a comment about bugs I found. Might try to check out code more carefully if you're having trouble fixing the bugs.

**Comment by frostyshadows** [comment]
@rupertotorres1 Thanks for testing, turns out delete was pretty messed up lol. The main issue was that our current implementation of deleting a PicNote didn't remove it from its folder, so the folder still had the deleted PicNote's ID in its list of PicNoteIDs.

- **When you add a picture to a new folder it looks weird. I don't know if it's supposed to be like this or we are going to change it, but the whole row turns grey and the folder id also spans the whole row.**
- It's based on Shavonne's designs, I'll check with her though.
- **I added two pictures to the same folder and deleted the one that appeared first. The remaining picture stayed in the center column, but I think it should move as far left as possible.**
- this should be fixed now
- **In the folder view, you can't enter the folder or press any picture in the folder, or delete folders or the pictures in them (you need to go to "All" to delete).**
- yep, it wasn't implemented in this ticket
- **After deleting all pictures in a folder, the folder view stays the same (pictures still appear) until I press take photo or add from library. After pressing take photo or add from library, the empty folder remains, but maybe it should be removed from the view if it has nothing in it.**
- this should also work now
- **Some of these bugs are probably also causing other bugs, because just by playing around I've had photos from one folder randomly appear in another folder (I've seen this happen when I add two pictures, remove the first one  and then add another picture to a new folder)**

**Review by vpnda — APPROVED** [review]

**Review by rupertotorres1 — APPROVED** [review]

**Pull request merged**